### PR TITLE
Aad workflow identity credentials

### PR DESCRIPTION
**Pull request by cccs-sgaron**

### CODE REVIEW

File: assemblyline_ui/api/v4/authentication.py [hunk 1]
```diff
@@ -56,6 +58,8 @@
 auth_api = make_subapi_blueprint(SUB_API, api_version=4)
 auth_api._doc = "Allow user to authenticate to the web server"
 
+TEMP_SECRET = "4mvvugJEuW=dA:3XqIDpHg2XqACWw/0-"
```
Comment (cccs-rs) [inline assemblyline_ui/api/v4/authentication.py:61]: Is this supposed to be here?

File: assemblyline_ui/api/v4/authentication.py [hunk 2]
```diff
@@ -663,28 +664,46 @@ def oauth_validate(**_):
...
+                            client_id=oauth_provider_config.client_id,
+                            tenant_id=oauth_provider_config.aad_wic_tenant_id,
+                            token_file_path=oauth_provider_config.aad_wic_token_file_path,
+                            additionally_allowed_tenants=oauth_provider_config.aad_wic_additionally_allowed_tenants)
+                    else:
+                        credentials = DefaultAzureCredential()
+
+                    if credentials:
+                        try:
+                            client_assertion = credentials.get_token(".default").token
```
Comment (ociappara) [inline assemblyline_ui/api/v4/authentication.py:693]: Might want to make the scope configurable?

File: assemblyline_ui/api/v4/authentication.py (hunk 1 above)
Comment (cccs-sgaron) [reply assemblyline_ui/api/v4/authentication.py]: Nope... good catch but I wont squash the commit as the client_id was not leaked anyway. . I deleted that secret out of extreme precaution anyway.

**Comment by sgaron-msft** [comment]
Can be removed, superseded by this PR: https://github.com/CybercentreCanada/assemblyline-ui/pull/1049

**Pull request merged**